packages/instantsearch-react/src/connectors/connectHierarchicalMenu.js

```diff
@@ -116,7 +116,7 @@ export default createConnector({
     const selectedItem = getSelectedItem(props, state);
     return {
       id,
-      filters: selectedItem === null ? [] : [{
+      filters: !selectedItem ? [] : [{
```

Comment:
The test on L104 should be the same as this one. Indeed, selectedItem can be either a non-empty string, in which case we consider that there is a selected item whose value is `selectedItem`, or an empty string or null, and in both cases we consider that there is no selected item. I'm not sure why I kept null as a possible value, as the default value should always be `''`.

The reason why the explicit absence of value must be represented as an empty string in state is because the state gets serialized to an URL:

* `{q: 'hello', category: null}` serializes to `?q=hello`, which deserializes to `{q: 'hello'}`: we lose information.
* `{q: 'hello', category: ''}` serializes to `?category=&q=hello`, which deserializes to the original `{q: 'hello', category: ''}`

So in the case where we set `category` to `null`, we lose the knowledge that the state of the `category` widget was explicitly reset. This only has consequences when the user of the library provides a default state for the `category` widget.

0. `const defaultState = {category: 'defaultCategory'};`
1. The state is initialized from the url.  
    `?q=hello` -> `{...defaultState, q: 'hello'}` -> `{q: 'hello', category: 'defaultCategory'}`
2. The user explicitly deselects the `defaultCategory` category.  
    `{q: 'hello', category: ''}` -> `'?category=&q=hello` -> `{...defaultState, q: 'hello', category: ''}` -> `{q: 'hello', category: ''}`

Does that make sense?

From the looks of it, this behaviour might be broken in the current version. I've updated this widget in a future PR (for tests), so maybe I should try to fix it there?